app/v2/notifications/rest_push.py

```diff
@@ -1,59 +1,106 @@
-        raise NotImplementedError()
-
     if not authenticated_service.has_permissions(PUSH_TYPE):
         raise BadRequestError(
-            message='Service is not allowed to send {}'.format(get_public_notify_type_text(PUSH_TYPE, plural=True))
+            message='Service is not allowed to send {}'.format(
+                get_public_notify_type_text(PUSH_TYPE, plural=True),
+            ),
+            status_code=403,
         )
 
-    req_json = validate(request.get_json(), schema)
+    validated_payload = validate_push_payload(schema)
 
-    if 'mobile_app' in req_json:
-        app_instance = mobile_app_registry.get_app(MobileAppType[req_json['mobile_app']])
-    else:
-        app_instance = mobile_app_registry.get_app(DEAFULT_MOBILE_APP_TYPE)
+    vetext_formatted_payload = vetext_client.format_for_vetext(validated_payload)
 
     try:
-        vetext_client.send_push_notification(
-            app_instance.sid,
-            req_json['template_id'],
-            req_json['topic_sid'] if is_broadcast else req_json['recipient_identifier']['id_value'],
-            is_broadcast,
-            req_json.get('personalisation'),
+        current_app.logger.debug(
+            'Attempting to call deliver_push celery task with validated payload: %s',
+            vetext_formatted_payload,
         )
-    except VETextBadRequestException as e:
-        raise BadRequestError(message=e.message, status_code=400)
-    except (VETextNonRetryableException, VETextRetryableException):
-        return jsonify(result='error', message='Invalid response from downstream service'), 502
+
+        # Choosing to use the email queue for push to limit the number of empty queues
+        deliver_push.apply_async(
+            args=(vetext_formatted_payload,),
+            queue=QueueNames.SEND_EMAIL,
+        )
+    except (CeleryError, OperationalError):
+        current_app.logger.exception('Failed to enqueue deliver_push request')
+        response = jsonify(result='error', message='VA Notify service impaired, please try again'), 502
     else:
-        return jsonify(result='success'), 201
+        response = jsonify(result='success'), 201
+    # Flask turns the tuple into a json and status_code
+    return response
+
+
+def validate_push_payload(schema: dict[str, str]) -> V2PushPayload:
+    """Validate an incoming push request.
+
+    Args:
+        schema (dict[str, str]): The incoming request
+
+    Raises:
+        BadRequestError: Failed validation
+
+    Returns:
+        dict[str, str]: Validated request dictionary
+    """
+    try:
+        req_json: dict[str, str] = validate(request.get_json(), schema)
+
+        # Validate the application they sent us is valid or use the default
+        if 'mobile_app' in req_json:
+            app_sid = mobile_app_registry.get_app(MobileAppType[req_json['mobile_app']]).sid
+        else:
+            app_sid = mobile_app_registry.get_app(DEAFULT_MOBILE_APP_TYPE).sid
```

Comment:
I know we've been using `DEAFULT_MOBILE_APP_TYPE` for the past four years (inherited typo) but since we are here, can we fix the spelling now, wherever it appears in the repo?

Reply:
Updating.